### PR TITLE
Graduating AppProtocol to GA

### PR DESCRIPTION
**Pull request by robscott**
**What type of PR is this?**
/kind feature
/kind api-change

**What this PR does / why we need it**:
This graduates the AppProtocol field on Services and Endpoints to GA.

**Does this PR introduce a user-facing change?**:
```release-note
AppProtocol is now GA for Endpoints and Services. The ServiceAppProtocol feature gate will be deprecated in 1.21.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
Enhancement Issue: https://github.com/kubernetes/enhancements/issues/1507

/sig network
/cc @bowei 
/assign @thockin 
/priority important-soon

### CODE REVIEW

File: pkg/controller/endpoint/endpoints_controller.go [hunk 1]
```diff
@@ -639,12 +639,10 @@ func shouldPodBeInEndpoints(pod *v1.Pod) bool {
...
-		Name:     servicePort.Name,
-		Port:     int32(portNum),
-		Protocol: servicePort.Protocol,
-	}
-	if utilfeature.DefaultFeatureGate.Enabled(features.ServiceAppProtocol) {
-		epp.AppProtocol = servicePort.AppProtocol
+		Name:        servicePort.Name,
+		Port:        int32(portNum),
+		Protocol:    servicePort.Protocol,
+		AppProtocol: servicePort.AppProtocol,
```
Comment (thockin) [inline pkg/controller/endpoint/endpoints_controller.go:645]: Do we not have this equivalent in EPSlice?  Why not?

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

File: pkg/controller/endpoint/endpoints_controller.go (hunk 1 above)
Comment (robscott) [reply pkg/controller/endpoint/endpoints_controller.go:645]: Good question - it actually predated all of this (v1.17). This feature gate guards a later addition of the same field to Services and Endpoints: https://github.com/kubernetes/kubernetes/blob/release-1.17/staging/src/k8s.io/api/discovery/v1alpha1/types.go#L142-L148.

**Comment by robscott** [comment]
@thockin I've fixed the failing tests - missed removing a feature gate check. PTAL.

**Pull request merged**